src/app/plan/duck/operations.ts

```diff
@@ -30,82 +30,170 @@ const pvFetchRequest = Creators.pvFetchRequest;
 const pvFetchFailure = Creators.pvFetchFailure;
 const pvFetchSuccess = Creators.pvFetchSuccess;
 const migrationSuccess = Creators.migrationSuccess;
+const stagingSuccess = Creators.stagingSuccess;
+const migrationFailure = Creators.migrationFailure;
+const stagingFailure = Creators.stagingFailure;
 const addPlanSuccess = Creators.addPlanSuccess;
-
-// const addPlanFailure = Creators.addPlanFailure;
-// const removePlanSuccess = Creators.removePlanSuccess;
-// const removePlanFailure = Creators.removePlanFailure;
 const sourceClusterNamespacesFetchSuccess = Creators.sourceClusterNamespacesFetchSuccess;
 
 const PollingInterval = 5000;
 const PvsDiscoveredType = 'PvsDiscovered';
 
+const groupPlan: any = (plan, response) => {
+  const fullPlan = {
+    MigPlan: plan.MigPlan,
+  };
+  if (response.data.items.length > 0) {
+    const sortMigrations = migrationList =>
+      migrationList.sort((left, right) => {
+        return moment
+          .utc(right.metadata.creationTimestamp)
+          .diff(moment.utc(left.metadata.creationTimestamp));
+      });
+
+    const matchingMigrations = response.data.items.filter(
+      i => i.kind === 'MigMigration' && i.spec.migPlanRef.name === plan.MigPlan.metadata.name
+    );
+
+    fullPlan['Migrations'] = sortMigrations(matchingMigrations);
+  } else {
+    fullPlan['Migrations'] = [];
+  }
+  return fullPlan;
+};
 const runStage = plan => {
-  return (dispatch, getState) => {
-    dispatch(Creators.initStage(plan.planName));
-    const planNameToStage = plan.planName;
-    const interval = setInterval(() => {
-      const planList = getState().plan.migPlanList;
-
-      const planItem = planList.find(p => p.planName === planNameToStage);
-      if (planItem.status.progress === 100) {
-        dispatch(Creators.stagingSuccess(planItem.planName));
-        clearInterval(interval);
+  return async (dispatch, getState) => {
+    try {
+      dispatch(Creators.initStage(plan.MigPlan.metadata.name));
+      dispatch(commonOperations.alertProgressTimeout('Staging Started'));
+      const { migMeta } = getState();
+      const client: IClusterClient = ClientFactory.hostCluster(getState());
+      const migMigrationObj = createMigMigration(
+        uuidv1(),
+        plan.MigPlan.metadata.name,
+        migMeta.namespace,
+        true
+      );
+      const migMigrationResource = new MigResource(MigResourceKind.MigMigration, migMeta.namespace);
+
+      //created migration response object
+      const createMigRes = await client.create(migMigrationResource, migMigrationObj);
+      const migrationListResponse = await client.list(migMigrationResource);
+      const groupedPlan = groupPlan(plan, migrationListResponse);
+
+      const callbackFunction = response => {
```

Comment:
I might give this a more descriptive name to indicate what its logic is actually doing.

Reply:
:+1: